src/binary-reader.cc

```diff
@@ -2028,7 +2039,7 @@ Result read_binary(const void* data,
   WABT_ZERO_MEMORY(logging_reader);
   logging_reader.user_data = &logging_context;
 
-  logging_reader.on_error = reader->on_error;
+  logging_reader.on_error = logging_on_error;
```

Comment:
OK, I've reworked the on_error handlers to return a bool which says whether the error was handled.